### PR TITLE
RBAC fix

### PR DESCRIPTION
**Pull request by margrs**
See https://github.com/ricoberger/vault-secrets-operator/pull/137

### CODE REVIEW

**Comment by amuraru** [comment]
The underlying issue seem to be this marker line in 
https://github.com/ricoberger/vault-secrets-operator/blob/main/controllers/vaultsecret_controller.go#L53


it needs to be split in two:
```golang
// +kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;watch;create;update;patch;delete
// +kubebuilder:rbac:groups=coordination.k8s.io,resources=leases,verbs=get;list;watch;create;update;patch;delete
```

so `controller-gen` does not override this in the future

**Comment by margrs** [comment]
> The underlying issue seem to be this marker line in https://github.com/ricoberger/vault-secrets-operator/blob/main/controllers/vaultsecret_controller.go#L53
> 
> it needs to be split in two:
> 
> ```go
> // +kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;watch;create;update;patch;delete
> // +kubebuilder:rbac:groups=coordination.k8s.io,resources=leases,verbs=get;list;watch;create;update;patch;delete
> ```
> 
> so `controller-gen` does not override this in the future

Thanks, @amuraru . Updated.

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -6,10 +6,20 @@ metadata:
   name: manager-role
 rules:
 - apiGroups:
-  - coordination.k8s.io
   - ""
   resources:
-  - configmaps
+  - leases
```
Comment (amuraru) [inline config/rbac/role.yaml:11]: This should be the other way around - leases not configmaps right?

Comment (margrs) [reply config/rbac/role.yaml]: Good catch, @amuraru . Updated.

**Review by amuraru — APPROVED** [review]

**Comment by ricoberger** [comment]
Hi @margrs and @amuraru, thanks for the fix and the review 🙂.

**Pull request merged**